### DISCUSSION TITLE
How to optimize x-ray tube simulation in egs++

### DISCUSSION THREAD

**Discussion by rtownson** [nrc-cnrc/EGSnrc#840] [Questions]
This question was originally posted here, and I have copied it over to preserve the discussion: https://www.reddit.com/r/EGSnrc/comments/sh1kwb/x_ray_tube_simulation_in_egs_through_toutor7pp/

> I created x ray tube and ran. But its took too much of cpu time even though I used UBS and parallel run in 8 core. its took 30 hours for deposit 3k particle in phasespace file. below i mentioned the trasnport parameter which is used in this run. How can i reduce time for 50kvp to 150kvp simulation

```
:start MC transport parameter:
    Global ECUT                     = 0.512 # Global electron transport cutoff
    Global PCUT                     = 0.001 # Global photon transport cutoff
    Global SMAX                     = 1e10 # Global maximum step-size restriction for e- transport
    ESTEPE                          = 0.25 # Default is 0.25
    XIMAX                           = 0.5 # Default is 0.5, max. value is 1.
    Boundary crossing algorithm     = exact # exact (default), PRESTA-I
    Skin depth for BCA              = 3 # Default value is 3 for exact boundary crossing
    Electron-step algorithm         = PRESTA-II # PRESTA-II (default),PRESTA-I
    Spin effects                    = On # On (default),Off
    Brems angular sampling          = KM # Simple,KM (default)
    Brems cross sections            = NIST # BH (default),NIST
    Pair angular sampling           = Simple # Off, Simple (default),KM
    Triplet production              = Off # On or Off (default)
    Electron Impact Ionization      = On # On, Off (default), casnati, kolbenstvedt, gryzinski
    Bound Compton scattering        = Simple # On, Off, Simple, norej (default)
    Radiative Compton corrections   = On # On, Off (default)
    Rayleigh scattering             = On # On ,Off (default), custom
    Photoelectron angular sampling  = On # On (default),Off
    Atomic relaxations              = On # On (default),Off
    Photon cross sections           = xcom # si, epdl, xcom (default) or user-supplied
    Photon cross-sections output    = Off # Off (default) or On
    Compton cross sections          = default # user-supplied
    Photonuclear attenuation        = On # Off (default) or On
    Photonuclear cross sections     = default # default (default) or user-supplied
:stop MC transport parameter:
```

The user also asks if it is OK to use `ECUT=AE=0.521`

**Comment by Jaap1957** [comment]
I would not use MC always to get xray spectra since they were measured already a long time ago.
So here a non EGS suggestion, although I very much appreciate EGS:

For Xray tube "simulation" one can refer to the CD-ROM published with this 1997 report, part of a publication of the British NHS: 

_Catalogue of Diagnostic X-ray Spectra and Other Data_, **IPEM Report 78**, K. Cranley, B. J. Gilmore, G. W. A. Fogarty and L. Desponds (Electronic version prepared by D. Sutton). The Institution of Physics and Engineering in Medicine and Biology. 1997. ISBN 090418188X

This program contains a large set of measured and simulated Xray tube spectra in the medical diagnostic range. See Simon Evan's 1998 [review](https://iopscience.iop.org/article/10.1088/0952-4746/18/1/026) for additional information.

Yes, this is a very old program. But still useful. It can be executed in windows 3.11 (!!) See
https://www.lifehacker.com.au/2019/03/how-to-install-ms-dos-6-22-and-windows-3-11-in-a-virtual-machine/
for how to install an old version of Windows on a virtual (time) machine.

**Reply by ftessier** [reply]
@Jaap1957 Interesting, thanks! But we cannot distribute the CD-ROM files here, so I have updated your post with a reference instead, and a reference to a review on it. Now this is quite old indeed, and it would be an interesting exercise to compare with modern calculations.
